src/Hyperion/TypeFilter.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Hyperion.Extensions;
+
+namespace Hyperion
+{
+    public sealed class TypeFilter : ITypeFilter
+    {
+        public ImmutableHashSet<string> FilteredTypes { get; }
+
+        internal TypeFilter(IEnumerable<Type> types)
+        {
+            FilteredTypes = types.Select(t => t.GetShortAssemblyQualifiedName()).ToImmutableHashSet();
+        }
+        
+        public bool IsAllowed(string typeName)
+            => FilteredTypes.Any(t => t == typeName);
+    }
+
+    internal sealed class DisabledTypeFilter : ITypeFilter
```

Comment:
Default ITypeFilter that always returns true, used when type filtering is disabled.